Math/NumberTheory/ArithmeticFunctions/Standard.hs

```diff
@@ -254,6 +257,33 @@ instance Monoid (Mangoldt a) where
   mempty  = MangoldtZero
   mappend = (<>)
 
+isNFree :: UniqueFactorisation n => Word -> n -> Bool
+isNFree n = runFunction (isNFreeA n)
+
+-- | Check if an integer is @n@-free. An integer @x@ is @n@-free if in its
+-- factorisation into prime factors, no factor has an exponent large than or
+-- equal to @n@.
+isNFreeA :: UniqueFactorisation n => Word -> ArithmeticFunction n Bool
+isNFreeA n = ArithmeticFunction (\_ pow -> All $ pow < n) getAll
+
+nFrees :: forall a . (Integral a, UniqueFactorisation a) => Word -> [a]
+nFrees n = concatMap nFreesListInternal nFreeList
+  where
+    stride :: Word
+    stride = 256
```

Comment:
Sieving is efficient only when the length of each individual sieve is at least sqrt of upper bound. We should either choose `stride` large enough, or take sieves of increasing size (256, 512, 1024, etc.).

Reply:
I'll address this afterwards.

Reply:
It sounds reasonable to me.